bugsnag-plugin-android-performance-okhttp/src/main/kotlin/com/bugsnag/android/performance/okhttp/BugsnagPerformanceOkhttp.kt

```diff
@@ -84,11 +91,50 @@ public class BugsnagPerformanceOkhttp : EventListener() {
         (spans.remove(call) as? SpanImpl)?.discard()
     }
 
-    public fun buildTraceparentHeader(
-        traceId: String,
-        parentSpanId: String,
+    private fun buildTraceParentHeader(
+        traceId: UUID,
+        parentSpanId: Long,
         sampled: Boolean,
     ): String {
-        return "00-$traceId-$parentSpanId-${if (sampled) "01" else "00"}"
+        return buildString {
+            append("00-")
+            appendHexUUID(traceId)
+            append('-')
+            appendHexLong(parentSpanId)
+            append('-')
+            append(if (sampled) "01" else "00")
+        }
     }
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val spanContext: SpanContext =
+            spans[chain.call()]
+                ?: SpanContext.current.takeUnless { it == SpanContext.invalid }
+                ?: return chain.proceed(chain.request())
+
+        val url = chain.request().url.toString()
+        if (!tracePropagationUrls.any { it.matcher(url).matches() }) {
+            return chain.proceed(chain.request())
+        }
```

Comment:
we could reduce the `return` count to 2 here (and get rid of the linter baseline):

```suggestion
        val spanContext: SpanContext = spans[chain.call()]
                ?: SpanContext.current.takeUnless { it == SpanContext.invalid }

        val url = chain.request().url.toString()
        if (spanContext == null || !tracePropagationUrls.any { it.matcher(url).matches() }) {
            return chain.proceed(chain.request())
        }
```